### PR TITLE
#11

### PR DESCRIPTION
**Pull request by RamaMahjoub**

### CODE REVIEW

File: migration/20221008202022-transaction-table.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+module.exports = class transactionTable20221008202022 {
+
+    async up(queryRunner) {
+        await queryRunner.query(
+            `create table \`transaction\`(
+                \`id\` char(36) not null,
+                \`transaction_date\` datetime not null,
+                \`transaction_type\` integer not null,
+                \`amount\` decimal(8,2) null,
```
Comment (foaudkajj) [inline migration/20221008202022-transaction-table.js]: amount should be not null.
@RamaMahjoub

File: src/models/transaction-card.model.ts [hunk 2]
```diff
@@ -31,11 +31,11 @@ export class TransactionCard {
   gsm: string;
 
   @Column({name: 'currency_id'})
-  @ApiProperty({required: true, format: 'uuid'})
+  @ApiProperty({required: false, format: 'uuid'})
```
Comment (foaudkajj) [inline src/models/transaction-card.model.ts]: currencyId is required. @RamaMahjoub

File: src/models/transaction.model.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    @ApiProperty({ required: true })
+    transactionDate: Date;
+
+    @Column({ name: 'transaction_type' })
+    @ApiProperty({ required: true, type: 'integer' })
+    transactionType: number;
+
+    @Column({ name: 'amount' })
+    @ApiProperty({ required: false, type: 'number', format: 'double' })
+    amount: number;
```
Comment (foaudkajj) [inline src/models/transaction.model.ts:28]: amount is required. @RamaMahjoub

File: migration/20221008202022-transaction-table.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                \`description\` varchar(1000) null,
+                \`payment_method_id\` char(36) null,
+                \`transaction_card_id\` char(36) null,
```
Comment (foaudkajj) [inline migration/20221008202022-transaction-table.js]: Sorry, I forgot to mention that. Please make `payment_method_id` and `transaction_card_id` required.
@RamaMahjoub

Comment (RamaMahjoub) [reply migration/20221008202022-transaction-table.js]: 👍🏻

**Pull request merged**